Source/Protocols/Request.swift

```diff
@@ -6,24 +6,32 @@ public typealias EmptyResponse = Void
 
 public protocol Request {
   typealias ResponseObject
+  typealias ResponseError = NSError
   func build() -> NSURLRequest
-  func parse(j: JSON) -> Result<ResponseObject, NSError>
+  func parse(j: JSON) -> Result<ResponseObject, SwishError>
+  func transformError(error: SwishError) -> ResponseError
```

Comment:
And yes, that's what this function is intended for. It mimics `parse` but for the error case.